### ISSUE TITLE
Error: Handshake request with invalid Connection header: 'keep-alive, Upgrade'

### ISSUE THREAD

**Issue by joepdooper** [sirn-se/websocket-php#39]
Hi there! Thank you for the library!

I am testing the library like so:

In the terminal:

```bash
php echoserver.php
```

In javascript:

```javascript
const socket = new WebSocket('ws://localhost:80');

socket.onopen = function(event) {
     console.log('WebSocket connected');
     socket.send('@info');
     socket.send('--opcode text "A text message"');
};

socket.onmessage = (event) => {
     console.log(event)
     document.getElementById('messages').innerHTML +=
          'Message from server: ' + event.data + "<br>";
     };

socket.onerror = function(error) {
     console.error('WebSocket error:', error);
};

socket.onclose = function(event) {
     console.log('WebSocket connection closed');
};
```

This works perfectly fine on Chrome and Safari, but on Firefox I get following errors

In the terminal:

Error: Handshake request with invalid Connection header: 'keep-alive, Upgrade'

In Firefox version 122.0.1 on GET:

426 Upgrade Required

As a newby on websockets I have been searching and trying to find out what causes this error but no luck so far…

**Comment by sirn-se** [comment]
Hi @joepdooper 

When client connects to server, the server expects the client to send a `Connection: upgrade` header. But apparently Firefox has it a bit different, `Connection: keep-alive, Upgrade`. I've never seen that before, and the keep-alive directive is unnecessary in this context.

But as I interpret the [RFC](https://datatracker.ietf.org/doc/html/rfc6455#section-4.1) above use should be valid. The requirement is that the header **include** the key phrase "Upgrade", but **may** also contain other key phrases.
I will update this repo with a fix for that asap.

**Comment by joepdooper** [comment]
Hi @sirn-se 

> I will update this repo with a fix for that asap.

That would be great! Thank you!

For the time being, while testing, I quickly "fixed it" in the library like so:

```php
if (stripos(strtolower($connectionHeader), 'upgrade') === false) {
     throw new HandshakeException(
               "Handshake request with invalid Connection header: '{$connectionHeader}'",
          $response->withStatus(426)
     );
}
```

**Comment by sirn-se** [comment]
Fixed in `v2.1.3`. Closing.
